### ISSUE TITLE
Can this project be implement in openpdet-v0.6 

### ISSUE THREAD

**Issue by RhythmOfTheRain-Byte** [CVMI-Lab/ST3D#75]
Excuse me，I would like to ask can this project be implement in openpdet-v0.6 ?
Thank you!

**Comment by RhythmOfTheRain-Byte** [comment]
Thank you for your reply When I run the test, the following problem occurs. I don't know what the problem is, can you help me to solve it?
The input command is as fellows
python tools/test.py --cfg_file OpenPCDet/tools/cfgs/da-nuscenes-kitti_models/pointrcnn_st3d/pointrcnn_st3d++_car.yaml --batch_size 2 --ckpt OpenPCDet/pretrained_models_st3d/st3d++/pointrcnn_st3d++_car_6751.pth

The terminal outputs the following error message：
2023-08-27 12:33:57,566   INFO  *************** Performance of EPOCH 6751 *****************
Traceback (most recent call last):
  File "/home/hyh/anaconda3/envs/openpcdet/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/hyh/anaconda3/envs/openpcdet/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/hyh/.vscode/extensions/ms-python.python-2023.4.1/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/__main__.py", line 39, in <module>
    cli.main()
  File "/home/hyh/.vscode/extensions/ms-python.python-2023.4.1/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/home/hyh/.vscode/extensions/ms-python.python-2023.4.1/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 284, in run_file
    runpy.run_path(target, run_name="__main__")
  File "/home/hyh/.vscode/extensions/ms-python.python-2023.4.1/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 322, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/home/hyh/.vscode/extensions/ms-python.python-2023.4.1/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 136, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/home/hyh/.vscode/extensions/ms-python.python-2023.4.1/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 124, in _run_code
    exec(code, run_globals)
  File "/home/hyh/OpenPCDet/tools_st3d/test.py", line 220, in <module>
    main()
  File "/home/hyh/OpenPCDet/tools_st3d/test.py", line 216, in main
    epoch_id, dist_test=dist_test)
  File "/home/hyh/OpenPCDet/tools_st3d/test.py", line 65, in eval_single_ckpt
    result_dir=eval_output_dir, save_to_file=args.save_to_file, args=args
  File "/home/hyh/OpenPCDet/tools_st3d/eval_utils/eval_utils.py", line 83, in eval_one_epoch
    sec_per_example = (time.time() - start_time) / len(dataloader.dataset)
ZeroDivisionError: float division by zero

**Comment by RhythmOfTheRain-Byte** [comment]
Excuse me， When I perform the training steps, the following error occurs, the yaml file used is 
“/tools/cfgs/da-nuscenes-kitti_models/pointrcnn_st3d/pointrcnn_st3d++_car.yaml”
I don't know how to solve this problem, could you  help me?
The terminal outputs the following error message

'DataAugmentor' object has no attribute 'random_object_scaling'
  File "/home/hyh/OpenPCDet/pcdet/datasets/augmentor/data_augmentor.py", line 24, in __init__
    cur_augmentor = getattr(self.augmentor_configs, cur_cfg.NAME)(config=cur_cfg)
  File "/home/hyh/OpenPCDet/pcdet/datasets/dataset.py", line 32, in __init__
    ) if self.training else None
  File "/home/hyh/OpenPCDet/pcdet/datasets/nuscenes/nuscenes_dataset.py", line 17, in __init__
    dataset_cfg=dataset_cfg, class_names=class_names, training=training, root_path=root_path, logger=logger
  File "/home/hyh/OpenPCDet/pcdet/datasets/__init__.py", line 60, in build_dataloader
    logger=logger,
  File "/home/hyh/OpenPCDet/tools/train.py", line 120, in main
    total_epochs=args.epochs
  File "/home/hyh/OpenPCDet/tools/train.py", line 246, in <module>
    main()
AttributeError: 'DataAugmentor' object has no attribute 'random_object_scaling'

**Comment by RhythmOfTheRain-Byte** [comment]
Excuse me, jihan.
I met another quertion when test the DENSE dataset using openpcdet-v0.6. May I  seek your assistance?
The details of the issue are described as follows:
I  train the dense dataset using OpenPCDet. However, when I use the trained model's .pth file for inference and run demo.py, it only displays half of the inference results. Changing the range of points doesn't seem to resolve the issue either. Which is shown as fellows. Could you please help me understand the reason behind this?
Additionally, the results obtained during the evaluation phase are as follows.The discrepancies between bbox AP, bev AP, and 3D AP are quite significant. I am not sure if this is reasonable. Which is as fellows:
Car AP@0.70, 0.70, 0.70:
bbox AP:81.2543, 81.2543, 81.2543
bev AP:32.0643, 32.0643, 32.0643
3d AP:21.4249, 21.4249, 21.4249
aos AP:58.71, 58.71, 58.71
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:82.4550, 82.4550, 82.4550
bev AP:28.3860, 28.3860, 28.3860
3d AP:15.9582, 15.9582, 15.9582
aos AP:59.12, 59.12, 59.12
Car AP@0.70, 0.50, 0.50:
bbox AP:81.2543, 81.2543, 81.2543
bev AP:43.2742, 43.2742, 43.2742
3d AP:40.7021, 40.7021, 40.7021
aos AP:58.71, 58.71, 58.71
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:82.4550, 82.4550, 82.4550
bev AP:40.8015, 40.8015, 40.8015
3d AP:37.2809, 37.2809, 37.2809
aos AP:59.12, 59.12, 59.12

![Screenshot from 2023-12-19 13-10-43](https://github.com/CVMI-Lab/ST3D/assets/79413024/75506809-0896-4235-ae7e-99c60a98004d)

**Comment by jihanyang** [comment]
Can you provide more information about your attempt?

**Comment by RhythmOfTheRain-Byte** [comment]
> Can you provide more information about your attempt?

Hello, thank you for your reply. I am conducting research on domain adaptation under different weather conditions. The DENSE dataset is needed because it contains some point cloud data under fog weather which is not available under other conditions.
Firstly, I would like to use some existing models to train on this dataset, and then take a look at the accuracy of the training results.
I used the pointpillar model to train on DENSE dataset. Then, using the Kitti evaluation criteria for evaluation, the problem described above was encountered. In addition, when I ran demo.py, I found that only a portion of point cloud targets could be detected which is shown as the above figure, when i change the POINT_CLOUD_RANGE, the result would not change.  Besides the difference between bbox AP,3d AP, bev AP is so big, I think this is unreasonable.
I don't know the reason, and seek for your help.

**Comment by jihanyang** [comment]
can you show me the config that you run?

**Comment by RhythmOfTheRain-Byte** [comment]
> can you show me the config that you run in demo.py?
Thank you for your reply,
I have the below four lines add to dense_dataset.yaml.
MAP_CLASS_TO_KITTI: {
    'Pedestrian': 'Pedestrian',
    'PassengerCar': 'Car',
    'RidableVehicle': 'Cyclist',
}
Besides i found other problem, the AP at different threshold is the same ,fow example:
Car AP@0.70, 0.70, 0.70: 
bbox AP:81.2543, 81.2543, 81.2543
bev AP:32.0643, 32.0643, 32.0643
3d AP:21.4249, 21.4249, 21.4249
aos AP:58.71, 58.71, 58.71

And the pointpillar.yaml is just as  fellows:
CLASS_NAMES: ['Pedestrian', 'PassengerCar', 'RidableVehicle'] 

DATA_CONFIG: 
    _BASE_CONFIG_: tools/cfgs/dataset_configs/dense_dataset.yaml
    POINT_CLOUD_RANGE: [-69.12, -39.68, -3, 69.12, 39.68, 1]
    # POINT_CLOUD_RANGE: [0, -39.68, -3, 69.12, 39.68, 1]

    DATA_PROCESSOR:
        - NAME: mask_points_and_boxes_outside_range
          REMOVE_OUTSIDE_BOXES: True

        - NAME: shuffle_points
          SHUFFLE_ENABLED: {
            'train': True,
            'test': False
          }

        - NAME: transform_points_to_voxels
          VOXEL_SIZE: [0.16, 0.16, 4]
          MAX_POINTS_PER_VOXEL: 32
          MAX_NUMBER_OF_VOXELS: {
            'train': 16000,
            'test': 40000
          }
    DATA_AUGMENTOR:
        DISABLE_AUG_LIST: ['placeholder']
        AUG_CONFIG_LIST:
            - NAME: gt_sampling
              USE_ROAD_PLANE: False
              DB_INFO_PATH:
                  - dense_dbinfos_train_clear.pkl
              PREPARE: {
                 filter_by_min_points: ['PassengerCar:5', 'Pedestrian:5', 'RidableVehicle:5'],
                 filter_by_difficulty: [-1],
              }

              SAMPLE_GROUPS: ['PassengerCar:15','Pedestrian:15', 'RidableVehicle:15']
              NUM_POINT_FEATURES: 5
              DATABASE_WITH_FAKELIDAR: False
              REMOVE_EXTRA_WIDTH: [0.0, 0.0, 0.0]
              LIMIT_WHOLE_SCENE: False

            - NAME: random_world_flip
              ALONG_AXIS_LIST: ['x']

            - NAME: random_world_rotation
              WORLD_ROT_ANGLE: [-0.78539816, 0.78539816]

            - NAME: random_world_scaling
              WORLD_SCALE_RANGE: [0.95, 1.05]

MODEL:
    NAME: PointPillar

    VFE:
        NAME: PillarVFE
        WITH_DISTANCE: False
        USE_ABSLOTE_XYZ: True
        USE_NORM: True
        NUM_FILTERS: [64]

    MAP_TO_BEV:
        NAME: PointPillarScatter
        NUM_BEV_FEATURES: 64

    BACKBONE_2D:
        NAME: BaseBEVBackbone
        LAYER_NUMS: [3, 5, 5]
        LAYER_STRIDES: [2, 2, 2]
        NUM_FILTERS: [64, 128, 256]
        UPSAMPLE_STRIDES: [1, 2, 4]
        NUM_UPSAMPLE_FILTERS: [128, 128, 128]

    DENSE_HEAD:
        NAME: AnchorHeadSingle
        CLASS_AGNOSTIC: False

        USE_DIRECTION_CLASSIFIER: True
        DIR_OFFSET: 0.78539
        DIR_LIMIT_OFFSET: 0.0
        NUM_DIR_BINS: 2

        ANCHOR_GENERATOR_CONFIG: [
            {
                'class_name': 'Pedestrian',
                'anchor_sizes': [[0.62, 0.6, 1.68]], ##数据的顺序:长宽高
                'anchor_rotations': [0, 1.57],
                # 'anchor_bottom_heights': [0.62], ##是否需要修改 Center_y - H/2  kitti [-0.6]
                'anchor_bottom_heights': [-0.6],
                'align_center': False,
                'feature_map_stride': 2,
                'matched_threshold': 0.5,
                'unmatched_threshold': 0.35
            },
            {
                'class_name': 'PassengerCar',
                'anchor_sizes': [[4.17, 1.85, 1.55]],  
                'anchor_rotations': [0, 1.57],
                # 'anchor_bottom_heights': [0.75],      ## kitti [-1.78]
                'anchor_bottom_heights': [-1.78],
                'align_center': False,
                'feature_map_stride': 2,
                'matched_threshold': 0.6,
                'unmatched_threshold': 0.45
            },
            {
                'class_name': 'RidableVehicle',
                'anchor_sizes': [[1.70,  0.67, 1.74]],
                'anchor_rotations': [0, 1.57],
                # 'anchor_bottom_heights': [0.62], ## kitti [-0.6]
                'anchor_bottom_heights': [-0.6],
                'align_center': False,
                'feature_map_stride': 2,
                'matched_threshold': 0.5,
                'unmatched_threshold': 0.35
            },
            # {
            #     'class_name': 'Car',
            #     'anchor_sizes': [[3.9, 1.6, 1.56]],
            #     'anchor_rotations': [0, 1.57],
            #     'anchor_bottom_heights': [-1.78],
            #     'align_center': False,
            #     'feature_map_stride': 2,
            #     'matched_threshold': 0.6,
            #     'unmatched_threshold': 0.45
            # },
            # {
            #     'class_name': 'Pedestrian',
            #     'anchor_sizes': [[0.8, 0.6, 1.73]],
            #     'anchor_rotations': [0, 1.57],
            #     'anchor_bottom_heights': [-0.6],
            #     'align_center': False,
            #     'feature_map_stride': 2,
            #     'matched_threshold': 0.5,
            #     'unmatched_threshold': 0.35
            # },
            # {
            #     'class_name': 'Cyclist',
            #     'anchor_sizes': [[1.76, 0.6, 1.73]],
            #     'anchor_rotations': [0, 1.57],
            #     'anchor_bottom_heights': [-0.6],
            #     'align_center': False,
            #     'feature_map_stride': 2,
            #     'matched_threshold': 0.5,
            #     'unmatched_threshold': 0.35
            # }
        ]

        TARGET_ASSIGNER_CONFIG:
            NAME: AxisAlignedTargetAssigner
            POS_FRACTION: -1.0
            SAMPLE_SIZE: 512
            NORM_BY_NUM_EXAMPLES: False
            MATCH_HEIGHT: False
            BOX_CODER: ResidualCoder

        LOSS_CONFIG:
            LOSS_WEIGHTS: {
                'cls_weight': 1.0,
                'loc_weight': 2.0,
                'dir_weight': 0.2,
                'code_weights': [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
            }

    POST_PROCESSING:
        RECALL_THRESH_LIST: [0.2, 0.25, 0.5]
        SCORE_THRESH: 0.1
        OUTPUT_RAW_SCORE: False

        EVAL_METRIC: kitti

        NMS_CONFIG:
            MULTI_CLASSES_NMS: False
            NMS_TYPE: nms_gpu
            NMS_THRESH: 0.2
            NMS_PRE_MAXSIZE: 4096
            NMS_POST_MAXSIZE: 500


OPTIMIZATION:
    BATCH_SIZE_PER_GPU: 4
    NUM_EPOCHS: 80

    OPTIMIZER: adam_onecycle
    LR: 0.003
    WEIGHT_DECAY: 0.01
    MOMENTUM: 0.9

    MOMS: [0.95, 0.85]
    PCT_START: 0.4
    DIV_FACTOR: 10
    DECAY_STEP_LIST: [35, 45]
    LR_DECAY: 0.1
    LR_CLIP: 0.0000001

    LR_WARMUP: False
    WARMUP_EPOCH: 1

    GRAD_NORM_CLIP: 10

**Comment by jihanyang** [comment]
Seems good. Do you train on `POINT_CLOUD_RANGE: [-69.12, -39.68, -3, 69.12, 39.68, 1]` or `POINT_CLOUD_RANGE: [0, -39.68, -3, 69.12, 39.68, 1]`? 
And can you try to increase: 
```
MAX_NUMBER_OF_VOXELS: {
      'train': 16000,
      'test': 40000
 }
```
You can refer the the value used in waymo or nuscenes.

**Comment by RhythmOfTheRain-Byte** [comment]
> Seems good. Do you train on `POINT_CLOUD_RANGE: [-69.12, -39.68, -3, 69.12, 39.68, 1]` or `POINT_CLOUD_RANGE: [0, -39.68, -3, 69.12, 39.68, 1]`? And can you try to increase:
> 
> ```
> MAX_NUMBER_OF_VOXELS: {
>       'train': 16000,
>       'test': 40000
>  }
> ```
> 
> You can refer the the value used in waymo or nuscenes.

Hello, jihan， sorry for the late response. I carefully reviewed my code and it seems I found an issue with the Average Precision (AP). It appears that the bounding boxes (BBOX) in the detection and the ones in INFOS are not both in the camera coordinate system, leading to a significant difference in AP. I corrected the corresponding errors, and the differences in AP values seem reasonable now. However, the AP is still not high. I want to know weather the result is affected by different datasets or not？
And the AP is as  fellows.
Pedestrian AP@0.50, 0.50, 0.50:
bbox AP:9.9371, 11.4342, 12.0241
bev  AP:5.5428, 5.7244, 5.8803
3d   AP:4.9156, 4.9627, 5.0251
aos  AP:6.36, 7.44, 7.67
Pedestrian AP_R40@0.50, 0.50, 0.50:
bbox AP:7.2823, 8.4742, 9.2234
bev  AP:1.5228, 1.8411, 1.9692
3d   AP:0.6597, 0.7109, 0.8487
aos  AP:4.46, 5.44, 5.80
Pedestrian AP@0.50, 0.25, 0.25:
bbox AP:9.9371, 11.4342, 12.0241
bev  AP:13.5843, 14.9021, 15.5474
3d   AP:13.2560, 14.3833, 14.9490
aos  AP:6.36, 7.44, 7.67
Pedestrian AP_R40@0.50, 0.25, 0.25:
bbox AP:7.2823, 8.4742, 9.2234
bev  AP:11.4294, 12.8239, 13.3732
3d   AP:10.8797, 12.1412, 12.7929
aos  AP:4.46, 5.44, 5.80
Car AP@0.70, 0.70, 0.70:
bbox AP:24.1654, 30.0964, 33.7385
bev  AP:26.2761, 30.6106, 33.0805
3d   AP:16.7971, 20.3112, 21.5931
aos  AP:18.63, 21.93, 23.77
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:22.8571, 28.2952, 31.3240
bev  AP:21.3982, 27.8406, 30.5966
3d   AP:10.6292, 14.8108, 16.7497
aos  AP:17.28, 20.31, 21.83
Car AP@0.70, 0.50, 0.50:
bbox AP:24.1654, 30.0964, 33.7385
bev  AP:36.2812, 43.0102, 46.6138
3d   AP:34.0325, 38.8539, 41.8615
aos  AP:18.63, 21.93, 23.77
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:22.8571, 28.2952, 31.3240
bev  AP:34.4048, 40.4873, 44.0209
3d   AP:31.2500, 37.0100, 40.4355
aos  AP:17.28, 20.31, 21.83
Cyclist AP@0.50, 0.50, 0.50:
bbox AP:13.5598, 13.5598, 14.9148
bev  AP:3.1287, 3.1287, 5.0066
3d   AP:2.4793, 2.4793, 5.0066
aos  AP:12.24, 12.24, 13.09
Cyclist AP_R40@0.50, 0.50, 0.50:
bbox AP:6.7711, 6.7711, 9.3816
bev  AP:1.9968, 1.9968, 3.2547
3d   AP:1.8182, 1.8182, 3.0435
aos  AP:5.30, 5.30, 7.22
Cyclist AP@0.50, 0.25, 0.25:
bbox AP:13.5598, 13.5598, 14.9148
bev  AP:14.3920, 14.3920, 14.9168
3d   AP:14.3920, 14.3920, 14.9168
aos  AP:12.24, 12.24, 13.09
Cyclist AP_R40@0.50, 0.25, 0.25:
bbox AP:6.7711, 6.7711, 9.3816
bev  AP:7.1068, 7.1068, 9.7445
3d   AP:7.1068, 7.1068, 9.7445
aos  AP:5.30, 5.30, 7.22

Additionally, when running the visualization demo.py, only the part where X < 0 is still displayed. Regarding the point cloud range, I made adjustments based on the annotated information of the dataset. Which is [0,-40.96,-6,128,40.96,4]. At the first training progress, I try the POINT_CLOUD_RANGE both as [-69.12, -39.68, -3, 69.12, 39.68, 1] and POINT_CLOUD_RANGE: [0, -39.68, -3, 69.12, 39.68, 1]. And neither of which can gain the bbox in the X<0 range.
The MAX_NUMBER_OF_VOXELS i set is as fellows, and i will try to increase the values.
          MAX_NUMBER_OF_VOXELS: {
            'train': 16000,
            'test': 40000
          }

Thanks for your reply!

**Comment by RhythmOfTheRain-Byte** [comment]
> Seems good. Do you train on `POINT_CLOUD_RANGE: [-69.12, -39.68, -3, 69.12, 39.68, 1]` or `POINT_CLOUD_RANGE: [0, -39.68, -3, 69.12, 39.68, 1]`? And can you try to increase:
> 
> ```
> MAX_NUMBER_OF_VOXELS: {
>       'train': 16000,
>       'test': 40000
>  }
> ```
> 
> You can refer the the value used in waymo or nuscenes.

Hello, jihan， sorry for the late response. I carefully reviewed my code and found an issue with the Average Precision (AP). It appears that the bounding boxes (BBOX) in the detection and the ones in INFOS are not both in the camera coordinate system, leading to a significant difference in AP. I corrected the corresponding errors, and the differences in AP values seem reasonable now. However, the AP is still not high.  As for the point cloud range, I made adjustments based on the annotated information of the dataset. Which is [0,-40.96,-6,128,40.96,4] 
I want to know weather the result is reasonable or not and is the low ap is related to the dataset itself?
And the AP is as fellows.
Pedestrian AP@0.50, 0.50, 0.50:
bbox AP:9.9371, 11.4342, 12.0241
bev AP:5.5428, 5.7244, 5.8803
3d AP:4.9156, 4.9627, 5.0251
aos AP:6.36, 7.44, 7.67
Pedestrian AP_R40@0.50, 0.50, 0.50:
bbox AP:7.2823, 8.4742, 9.2234
bev AP:1.5228, 1.8411, 1.9692
3d AP:0.6597, 0.7109, 0.8487
aos AP:4.46, 5.44, 5.80
Pedestrian AP@0.50, 0.25, 0.25:
bbox AP:9.9371, 11.4342, 12.0241
bev AP:13.5843, 14.9021, 15.5474
3d AP:13.2560, 14.3833, 14.9490
aos AP:6.36, 7.44, 7.67
Pedestrian AP_R40@0.50, 0.25, 0.25:
bbox AP:7.2823, 8.4742, 9.2234
bev AP:11.4294, 12.8239, 13.3732
3d AP:10.8797, 12.1412, 12.7929
aos AP:4.46, 5.44, 5.80
Car AP@0.70, 0.70, 0.70:
bbox AP:24.1654, 30.0964, 33.7385
bev AP:26.2761, 30.6106, 33.0805
3d AP:16.7971, 20.3112, 21.5931
aos AP:18.63, 21.93, 23.77
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:22.8571, 28.2952, 31.3240
bev AP:21.3982, 27.8406, 30.5966
3d AP:10.6292, 14.8108, 16.7497
aos AP:17.28, 20.31, 21.83
Car AP@0.70, 0.50, 0.50:
bbox AP:24.1654, 30.0964, 33.7385
bev AP:36.2812, 43.0102, 46.6138
3d AP:34.0325, 38.8539, 41.8615
aos AP:18.63, 21.93, 23.77
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:22.8571, 28.2952, 31.3240
bev AP:34.4048, 40.4873, 44.0209
3d AP:31.2500, 37.0100, 40.4355
aos AP:17.28, 20.31, 21.83
Cyclist AP@0.50, 0.50, 0.50:
bbox AP:13.5598, 13.5598, 14.9148
bev AP:3.1287, 3.1287, 5.0066
3d AP:2.4793, 2.4793, 5.0066
aos AP:12.24, 12.24, 13.09
Cyclist AP_R40@0.50, 0.50, 0.50:
bbox AP:6.7711, 6.7711, 9.3816
bev AP:1.9968, 1.9968, 3.2547
3d AP:1.8182, 1.8182, 3.0435
aos AP:5.30, 5.30, 7.22
Cyclist AP@0.50, 0.25, 0.25:
bbox AP:13.5598, 13.5598, 14.9148
bev AP:14.3920, 14.3920, 14.9168
3d AP:14.3920, 14.3920, 14.9168
aos AP:12.24, 12.24, 13.09
Cyclist AP_R40@0.50, 0.25, 0.25:
bbox AP:6.7711, 6.7711, 9.3816
bev AP:7.1068, 7.1068, 9.7445
3d AP:7.1068, 7.1068, 9.7445
aos AP:5.30, 5.30, 7.22

Additionally, when running the demo.py,  the part where X < 0 is still not displayed. At the first training progress, I try the POINT_CLOUD_RANGE both as [-69.12, -39.68, -3, 69.12, 39.68, 1] and POINT_CLOUD_RANGE: [0, -39.68, -3, 69.12, 39.68, 1]. And neither of which can gain the bboxes in the X<0 range.
The MAX_NUMBER_OF_VOXELS i set is as fellows, and i will try to increase the values.
MAX_NUMBER_OF_VOXELS: {
'train': 16000,
'test': 40000
}

Thanks for your reply!

**Comment by RhythmOfTheRain-Byte** [comment]
> Seems good. Do you train on `POINT_CLOUD_RANGE: [-69.12, -39.68, -3, 69.12, 39.68, 1]` or `POINT_CLOUD_RANGE: [0, -39.68, -3, 69.12, 39.68, 1]`? And can you try to increase:
> 
> ```
> MAX_NUMBER_OF_VOXELS: {
>       'train': 16000,
>       'test': 40000
>  }
> ```
> 
> You can refer the the value used in waymo or nuscenes.

Hello, jihan， sorry for the late response. I carefully reviewed my code and found an issue with the Average Precision (AP). It appears that the bounding boxes (BBOX) in the detection and the ones in INFOS are not both in the camera coordinate system, leading to a significant difference in AP. I corrected the corresponding errors, and the differences in AP values seem reasonable now. However, the AP is still not high. As for the point cloud range, I made adjustments based on the annotated information of the dataset. Which is [0,-40.96,-6,128,40.96,4]
I want to know weather the result is reasonable or not and is the low ap is related to the dataset itself?
And the AP is as fellows.
Pedestrian AP@0.50, 0.50, 0.50:
bbox AP:16.9423, 18.3138, 17.8672
bev  AP:11.8912, 12.5717, 12.9226
3d   AP:10.7305, 11.2102, 11.3135
aos  AP:8.57, 9.54, 9.21
Pedestrian AP_R40@0.50, 0.50, 0.50:
bbox AP:10.8295, 12.4563, 12.5069
bev  AP:4.7797, 5.5686, 5.6957
3d   AP:3.3617, 4.1809, 4.1089
aos  AP:6.83, 7.69, 7.73
Pedestrian AP@0.50, 0.25, 0.25:
bbox AP:16.9423, 18.3138, 17.8672
bev  AP:17.1872, 18.7117, 19.4163
3d   AP:16.6205, 18.2797, 18.9800
aos  AP:8.57, 9.54, 9.21
Pedestrian AP_R40@0.50, 0.25, 0.25:
bbox AP:10.8295, 12.4563, 12.5069
bev  AP:10.8060, 12.8820, 13.2827
3d   AP:10.2664, 12.1865, 12.7310
aos  AP:6.83, 7.69, 7.73
Car AP@0.70, 0.70, 0.70:
bbox AP:32.1993, 38.7883, 42.0663
bev  AP:34.2908, 40.6023, 42.9996
3d   AP:20.4010, 24.7456, 25.5088
aos  AP:24.67, 28.81, 30.90
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:28.3628, 35.6259, 38.8448
bev  AP:31.1468, 38.0147, 39.8974
3d   AP:15.5781, 20.2444, 21.6306
aos  AP:20.69, 25.44, 27.39
Car AP@0.70, 0.50, 0.50:
bbox AP:32.1993, 38.7883, 42.0663
bev  AP:45.7023, 53.5950, 56.8036
3d   AP:42.2839, 50.3538, 53.1686
aos  AP:24.67, 28.81, 30.90
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:28.3628, 35.6259, 38.8448
bev  AP:43.8367, 53.0522, 56.2826
3d   AP:39.9097, 48.4991, 51.2990
aos  AP:20.69, 25.44, 27.39
Cyclist AP@0.50, 0.50, 0.50:
bbox AP:23.9410, 22.1417, 21.6978
bev  AP:23.2225, 20.8878, 19.5872
3d   AP:18.2245, 15.6316, 15.4828
aos  AP:17.18, 16.06, 15.37
Cyclist AP_R40@0.50, 0.50, 0.50:
bbox AP:22.0832, 20.2484, 19.4745
bev  AP:19.6143, 18.0734, 17.1456
3d   AP:15.4045, 14.6243, 13.6611
aos  AP:15.90, 14.64, 13.65
Cyclist AP@0.50, 0.25, 0.25:
bbox AP:23.9410, 22.1417, 21.6978
bev  AP:24.2363, 22.5298, 22.0752
3d   AP:24.2298, 22.2064, 21.7664
aos  AP:17.18, 16.06, 15.37
Cyclist AP_R40@0.50, 0.25, 0.25:
bbox AP:22.0832, 20.2484, 19.4745
bev  AP:22.2573, 20.5275, 19.7453
3d   AP:22.2097, 20.3773, 19.5598
aos  AP:15.90, 14.64, 13.65

Additionally, when running the demo.py, the part where X < 0 is still not displayed. At the first training progress, I try the POINT_CLOUD_RANGE both as [-69.12, -39.68, -3, 69.12, 39.68, 1] and POINT_CLOUD_RANGE: [0, -39.68, -3, 69.12, 39.68, 1]. And neither of which can gain the bboxes in the X<0 range.
The MAX_NUMBER_OF_VOXELS i set is as fellows, and i will try to increase the values.
MAX_NUMBER_OF_VOXELS: {
'train': 16000,
'test': 40000
}

Thanks for your reply!
